### PR TITLE
Print icd data only when it is really needed

### PR DESCRIPTION
**Pull request by willemferguson**
Reduce the impact and visibility of icd calculations by writing
the data to the planner output only in cases where gas switches
involve:

1) ascent
2) helium is used as part of breathing gas
3) gas switch results in an increase in partial pressure of nitrogen.

Coding change:

1) The code for writing the header of the icd table as well as
   the inclusion of the data in the planner output is made
   conditional on the above three requirements. This involves
   moving the code that writes the table header into the
   function that writes an icd data item to buffer.
2) Creating a boolean variable that controls writing
   of the table header to buffer and that also determines
   whether any icd results should be included in the planner
   results at all.

Signed-off-by: Willem Ferguson <willemferguson@zoology.up.ac.za>

### Describe the pull request:

- [ ] Bug fix
- [x] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Mentions:

@atdotde 
@neolit123 
@sfuchs79

### CODE REVIEW

File: core/plannernotes.c [hunk 1]
```diff
@@ -536,7 +537,7 @@ void add_plan_to_notes(struct diveplan *diveplan, struct dive *dive, bool show_d
 	}
 
 	/* For trimix OC dives, add the ICD table here */
-	if (istrimix) {
+	if (!icdtableheader) {					// If an icd table header has been printed to buffer, then ..
```
Comment (neolit123) [inline core/plannernotes.c]: probably best to move this comment close to the `{`

File: core/plannernotes.c [hunk 2]
```diff
@@ -536,7 +537,7 @@ void add_plan_to_notes(struct diveplan *diveplan, struct dive *dive, bool show_d
...
 		icdlen += snprintf(icdbuffer + icdlen, sz_icdbuf - icdlen,"</tbody></table>");	// End the ICD table
 		len += snprintf(buffer + len, sz_buffer - len, "%s", icdbuffer);				// ..and add it to the html buffer
 		if (icdwarning) {																// If necessary, add a warning message to html buffer
```
Comment (neolit123) [inline core/plannernotes.c]: same for the above comments - these seem to use tabs and should be probably one space away from the code. you can add a separate patch on top for them.

**Review by neolit123 — APPROVED** [review]
we prefer whitespace fixes to be into a separate commit/patch, but given there is a comment about that in the commit message:
> 3) Reorganise comments so that there are not pultiple tabs between
   the code and the start of a comment.

its OK. lgtm.

**Comment by willemferguson** [comment]
Is there anything that I need to do?

**Pull request merged**